server/moderation/azure/azure.go

```diff
@@ -0,0 +1,166 @@
+package azure
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/mattermost/mattermost-plugin-content-moderator/server/moderation"
+	"github.com/pkg/errors"
+)
+
+const (
+	// ContentSafetyTextAnalyzeEndpoint is the Azure AI Content Safety text analyze API path
+	ContentSafetyTextAnalyzeEndpoint = "/contentsafety/text:analyze?api-version=2024-09-01"
```

Comment:
Should this be configurable?  or at least the version?

Reply:
I don't think the version should be configurable. If we let the end user configure this, they may pick a newer version with breaking changes to the API. By hardcoding it, we ensure the code and the API version stay in sync. WDYT?

Reply:
I'm thinking we may need to support multiple versions at once? We could keep a hardcoded list of supported versions in the plugin and they can choose it? I might be overthinking it though, I'm 0/5 on what you want to do with this

Reply:
Yeah for sure!